applications/main/nfc/plugins/supported_cards/saflok.c

```diff
@@ -153,13 +253,172 @@ static bool saflok_read(Nfc* nfc, NfcDevice* device) {
     return is_read;
 }
 
+bool saflok_parse(const NfcDevice* device, FuriString* parsed_data) {
+    furi_assert(device);
+    const MfClassicData* data = nfc_device_get_data(device, NfcProtocolMfClassic);
+    bool parsed = false;
+
+    do {
+        // Check card type
+        if(data->type != MfClassicType1k) break;
+
+        // Verify key
+        const MfClassicSectorTrailer* sec_tr =
+            mf_classic_get_sector_trailer_by_sector(data, CHECK_SECTOR);
+
+        const uint64_t key_a =
+            bit_lib_bytes_to_num_be(sec_tr->key_a.data, COUNT_OF(sec_tr->key_a.data));
+        if(key_a != saflok_1k_keys[CHECK_SECTOR].a) break;
+
+        // Decrypt basic access
+        uint8_t basicAccess[BASIC_ACCESS_BYTE_NUM];
+        for(int i = 0; i < 16; i++) {
+            basicAccess[i] = data->block[1].data[i];
+        }
```

Comment:
`memcpy()` ? i know i'm nitpicking at this point xD looks and works great from what i can see

Reply:
```suggestion
        uint8_t basicAccess[BASIC_ACCESS_BYTE_NUM];
        memcpy(basicAccess, data->block[1].data, 16);
```
seems to be working fine here

Reply:
It might need to be `&basicAccess` but yeah. I'm compiling just to be sure. Will push as soon as it works.